### PR TITLE
Feature/request driver

### PR DESCRIPTION
**Pull request by TheNaoX**
* Add feature file for requesting a driver
* Add controller and model for driver requests
* Add request creator service with specs
* Add time restriction to request
* Add expiration to response for requests controller
* Replace `.empty?` with `.any?` for request model class method
* Change service arguments for keyword arguments
* Change the `.empty?` to `.any?` on requests creator service

### CODE REVIEW

File: app/controllers/api/users/requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module Api
+  module Users
+    class RequestsController < ::Api::Users::BaseController
+      def create
+        ::Users::Requests::Create.(user,
+                                   request_params[:latitude],
+                                   request_params[:longitude],
+                                   success: method(:create_success),
+                                   error: method(:create_error))
```
Comment (cesargomez89) [inline app/controllers/api/users/requests_controller.rb]: Sending more than 4 parameters we are forcing to the devs to remember the arguments position.
Also the line becomes to long and you need to indent the code in this way.
a good way to deal with this, is sending a Hash {a: 'value' } so we don't care about the position.

Sandi Metz said something about this in his book or rules I bet

Comment (TheNaoX) [inline app/controllers/api/users/requests_controller.rb]: And it raises errors if values are present as well :smile:

File: app/models/request.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Request < ActiveRecord::Base
+  belongs_to :user
+
+  validates :latitude, :longitude, presence: true
+
+  before_create :set_expiration_date
+
+  def self.has_active_requests?(user)
+    not where(user: user,
+              expires_at: Time.current..3.minutes.from_now).empty?
```
Comment (cesargomez89) [inline app/models/request.rb]:
```ruby 
where(user: user, expires_at: Time.current..3.minutes.from_now).any?
```
or

```ruby 
!!find_by(user: user, expires_at: Time.current..3.minutes.from_now)
```
any? is the opposite of empty?

the second one is a where but just gets the first record
you can use !! to get a real boolen

Comment (TheNaoX) [inline app/models/request.rb]: I guess I'll go for the `any?` I didn't remember that one

File: app/services/users/requests/create.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          success.call(request)
+        else
+          error.call("no_drivers_around")
+        end
+      end
+
+      private
+
+      def drivers_around?
+        not Driver.geosearch(latitude, longitude).empty?
```
Comment (cesargomez89) [inline app/services/users/requests/create.rb]:
```
.any?
```

Comment (TheNaoX) [inline app/services/users/requests/create.rb]: :+1:

File: app/services/users/requests/create.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      end
+
+      def request
+        @request ||= Request.create_request(user, latitude, longitude)
```
Comment (cesargomez89) [inline app/services/users/requests/create.rb]: why do you need create_request() ?
I mean is doing the same as create()

**Comment by cesargomez89** [comment]
I left some comments @TheNaoX

File: app/services/users/requests/create.rb (hunk 4 above)
Comment (TheNaoX) [inline app/services/users/requests/create.rb]: That's for defining the entry point for the model, and be very specific on the dependencies. I don't want a bunch of explicit `.create` calls to be messing with the data model, I'm redefining the model's specific API for that.

**Comment by TheNaoX** [comment]
@cesargomez89 check again :)

File: app/controllers/api/users/requests_controller.rb (hunk 1 above)
Comment (cesargomez89) [inline app/controllers/api/users/requests_controller.rb]: nice!, thanks for clarify

**Pull request merged**